syft/frameworks/torch/he/fv/decrypter.py

```diff
@@ -0,0 +1,48 @@
+from syft.frameworks.torch.he.fv.plaintext import PlainText
+from syft.frameworks.torch.he.fv.util.operations import get_significant_count
+
+
+class Decrypter:
+    """Decrypts Ciphertext objects into Plaintext objects. Constructing a Decryptor
+    requires a Context object with valid encryption parameters, and the secret key.
+    """
+
+    def __init__(self, context, secret_key):
+        self._context = context
+        self._coeff_modulus = context.param.coeff_modulus
+        self._coeff_count = context.param.poly_modulus_degree
+        self._secret_key = secret_key.data
+
+    def decrypt(self, encrypted):
+        """Decrypts the encrypted ciphertext objects and return plaintext object
+
+        Args:
+            encrypted: A ciphertext object which has to be decrypted.
+        """
+
+        # Calculate [c0 + c1 * sk]_q
+        temp_product_modq = self.dot_product_ct_sk(encrypted)
+
+        # Divide scaling variant using BEHZ FullRNS techniques
+        result = self._context.rns_tool.decrypt_scale_and_round(temp_product_modq)
+
+        # removing leading zeroes in plaintext representation.
+        plain_coeff_count = get_significant_count(result)
+        return PlainText(result[:plain_coeff_count])
+
+    def dot_product_ct_sk(self, encrypted):
+        """calculate and return the value of [c0 + c1 * sk]_q
+        where [c0, c1] denotes encrypted ciphertext and sk is secret key.
+        """
+        phase = [0] * self._coeff_count * len(self._coeff_modulus)
+        c_0, c_1 = encrypted.data
+
+        for j in range(len(self._coeff_modulus)):
+            for i in range(self._coeff_count):
+                phase[i + j * self._coeff_count] = (
+                    (c_1[i + j * self._coeff_count] * self._secret_key[i + j * self._coeff_count])
+                    % self._coeff_modulus[j]
+                    + c_0[i + j * self._coeff_count]
+                ) % self._coeff_modulus[j]
```

Comment:
https://github.com/microsoft/SEAL/blob/f7d748c97ed841376c4a1cdec9e7c978f5e64a95/native/src/seal/decryptor.cpp#L101

Reply:
That's because they are using NTT. We have two choices here:

1. Use NTT and let it as is
2. Switch to polynomial operations

In both cases it would be helpful to have operations like addition and multiplication between two ring elements (either they are NTT transformed or not) be separated into functions and used across the code base so that updating to NTT isn't a big deal.

Reply:
Ok, I add NTT transformation👍